packages/iris-grid/src/IrisGridTreeTableModel.ts

```diff
@@ -15,18 +27,58 @@ class IrisGridTreeTableModel extends IrisGridTableModelTemplate<
   TreeTable,
   UITreeRow
 > {
-  // table: TreeTable
+  /** We keep a virtual column at the front that tracks the "group" that is expanded */
+  private virtualColumns: Column[];
+
+  constructor(
+    dh: DhType,
+    table: TreeTable,
+    formatter = new Formatter(dh),
+    inputTable: InputTable | null = null
+  ) {
+    super(dh, table, formatter, inputTable);
+    this.virtualColumns = [
+      {
+        name: 'Group',
+        type: TableUtils.dataType.STRING,
+        constituentType: TableUtils.dataType.STRING,
+        isPartitionColumn: false,
+        isSortable: false,
+        description: 'Virtual column',
+        filter: () => {
+          throw new Error('Filter not implemented for virtual column');
+        },
+        sort: () => {
+          throw new Error('Sort not implemented virtual column');
+        },
+        formatColor: () => {
+          throw new Error('Color not implemented for virtual column');
+        },
+        formatRowColor: () => {
+          throw new Error('Color not implemented for virtual column');
+        },
+      },
+    ];
+  }
 
   applyBufferedViewport(
     viewportTop: number,
     viewportBottom: number,
     columns: Column[]
   ): void {
-    this.table.setViewport(viewportTop, viewportBottom, columns);
+    const viewportColumns = [
+      // Need to always fetch the grouped columns
+      ...this.table.groupedColumns,
+      ...columns.filter(
+        c =>
+          !this.virtualColumns.includes(c) && !this.groupedColumns.includes(c)
```

Comment:
Yes `columns` is just whichever columns should be visible. Yes if `groupedColumns` is very large, we're pretty much hosed. Isn't really a way to do this virtual column without getting all the key column data as well.
Checking in the engine, it seems like it can't really handle >100 columns anyway, so I think we should be fine even in the maximum case the engine can handle. I tested using a snippet like this:
```
from deephaven import empty_table

column_count = 120
row_count = 1000
group_count = 50 # Increase this to 80 to see an error

t = empty_table(row_count).update("x=i")
for c in range(1,column_count):
    t = t.update(f"x{c}=i*{c}")

by = ["x"]
for i in range(1, group_count):
    by = by + [f"x{i}"]
r = t.rollup(aggs=[], by=by, include_constituents=True)
```

I'm going to ask in the DHC channel to see if it's expected to be able to handle more grouping columns.

Reply:
Looks like core has a ticket open for that issue: https://github.com/deephaven/deephaven-core/issues/4758
But it's backlogged as it doesn't seem like something users actually want to do.